app/jobs/backorder_job.rb

```diff
@@ -55,11 +55,16 @@ def place_backorder(order)
       ordered_quantities[item] = retail_quantity
     end
 
+    return if backorder.lines.empty?
+
     place_order(user, order, orderer, backorder)
 
     items.each do |item|
       variant = item.variant
-      variant.on_hand += ordered_quantities[item] if variant.on_demand
+      quantity = ordered_quantities[item]
+      next if quantity.zero?
+
+      variant.on_hand += quantity if variant.on_demand
```

Comment:
Wouldn't it be just as efficient to let `variant.on_hand += 0` run, rather than adding a check?

Reply:
No, it triggers several update statements.